imageio/imageio-dds/src/main/java/com/twelvemonkeys/imageio/plugins/dds/DX10Header.java

```diff
@@ -0,0 +1,29 @@
+package com.twelvemonkeys.imageio.plugins.dds;
+
+import javax.imageio.stream.ImageInputStream;
+import java.io.IOException;
+
+//https://learn.microsoft.com/en-us/windows/win32/direct3ddds/dds-header-dxt10
+public final class DX10Header {
+    final DX10DXGIFormat dxgiFormat;
+    final int resourceDimension, miscFlag, arraySize, miscFlags2;
+
+    private DX10Header(int dxgiFormat, int resourceDimension, int miscFlag, int arraySize, int miscFlags2) {
+        this.dxgiFormat = DX10DXGIFormat.getFormat(dxgiFormat);
+        this.resourceDimension = resourceDimension;
+        //only support D3D10_RESOURCE_DIMENSION_TEXTURE2D = 3
+        if (this.resourceDimension != 3) throw new UnsupportedOperationException("Resource dimension " + resourceDimension + " is not supported");
```

Comment:
I think it would be better if we add a constant for `D3D10_RESOURCE_DIMENSION_TEXTURE2D` either in the `DDS` interface,  or just local this class.